src/librbd/api/Namespace.cc

```diff
@@ -155,6 +155,34 @@ int Namespace<I>::list(IoCtx& io_ctx, vector<string> *names)
   return 0;
 }
 
+template <typename I>
+int Namespace<I>::exists(librados::IoCtx& io_ctx, const std::string& name, bool *exists)
+{
+  CephContext *cct = (CephContext *)io_ctx.cct();
+  ldout(cct, 5) << "name=" << name << dendl;
+
+  if (name.empty()) {
+    return -EINVAL;
+  }
+
+  librados::IoCtx ns_ctx;
+  ns_ctx.dup(io_ctx);
+  ns_ctx.set_namespace(name);
+
+  int r = librbd::cls_client::dir_state_assert(&ns_ctx, RBD_DIRECTORY,
+                                               cls::rbd::DIRECTORY_STATE_READY);
+  if (r == 0) {
```

Comment:
@dillaman  Probably, it should be `r >= 0`, right?

Reply:
Eh -- that will only ever return zero on success.

Reply:
Got it, thank you.